src/main/java/org/c02e/jpgpj/JCAContextHelper.java

```diff
@@ -0,0 +1,111 @@
+package org.c02e.jpgpj;
+
+import org.bouncycastle.gpg.keybox.jcajce.JcaKeyBoxBuilder;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.operator.PGPDigestCalculatorProvider;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPContentVerifierBuilderProvider;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPDigestCalculatorProviderBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePBEDataDecryptorFactoryBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePGPDataEncryptorBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyDataDecryptorFactoryBuilder;
+
+import java.security.Provider;
+import java.security.Security;
+
+/**
+ * Helper class for Java Cryptography Architecture (JCA) context
+ * consisting of {@link java.security.Provider provider}
+ * Note: The following class is not thread safe, the security provider should not be changed during PGP operations
+ */
+public class JCAContextHelper {
```

Comment:
Let's name this class `JcaContextHelper` instead of `JCAContextHelper`, to match the casing convention of other class names with initialisms in this project, like `EncryptedAsciiArmorHeadersCallback` etc.